bbin/git.py

```diff
@@ -3,10 +3,12 @@
 from pathlib import Path
 from typing import Any, Optional
 
-from . import utils
+from . import utils, interface
 
 GIT = shutil.which("git") or str(Path("/usr/bin/git"))
 
+if not GIT:
+    interface.warn("You do not have git installed!")
 
```

Comment:
I think we should use interface.error and fail entirely

Reply:
yeah

Reply:
Oh and also, we already have a pathlib callback. You might want to remove that...

Reply:
Should we also add a call to os.envrion.get?

Reply:
Since we *have* a callback, maybe we should warn the user of the callback

Reply:
I see.
try this?
```python3
GIT = shutil.which("git")
if GIT is None:
    raise click.ClickException("You do not have git installed!")
```

Reply:
I like the callback. Maybe a friendly warning utilizing interface.warn? Warn them that the callback will be used

Reply:
So... keep the callback, and warn them if it will be used.

```python3
GIT = shutil.which("git")
if GIT is None:
    interface.warn("You do not have git installed! BBin will use //usr//bin//git")

Reply:
Yes! Exactly!
```py
GIT = shutil.which("git")
if GIT is None:
    callback = Path(os.getenv("BBIN_GIT_CALLBACK", "/usr/bin/git"))  # NOTE: Dynamically change callback depending on Windows....
    if os.getenv("BBIN_NO_WARN_GIT") != "1":
        interface.warn(f"You do not have git installed! BBin will use the default `{callback}` callback you can change this by setting `BBIN_GIT_CALLBACK` to your Git executable\n(You can disable this warning in the future by setting `BBIN_NO_WARN_GIT` to `1`)")

    
```

Reply:
Why not ask upfront if the user wants to use the callback.

Reply:
I would like the installation process to be more automatic.

Reply:
Understood.